dask/dataframe/io/io.py

```diff
@@ -646,24 +650,25 @@ def to_records(df):
     return df.map_partitions(M.to_records)
 
 
-# TODO: type this -- causes lots of papercuts
 @insert_meta_param_description
 def from_delayed(
-    dfs,
+    dfs: Union[Delayed, Iterable[Union[Delayed, distributed.Future]]],
     meta=None,
-    divisions=None,
-    prefix="from-delayed",
-    verify_meta=True,
+    divisions=Union[tuple, Literal["sorted"], None],
+    prefix: str = "from-delayed",
+    verify_meta: bool = True,
 ):
     """Create Dask DataFrame from many Dask Delayed objects
 
     Parameters
     ----------
-    dfs : list of Delayed or Future
-        An iterable of ``dask.delayed.Delayed`` objects, such as come from
-        ``dask.delayed`` or an iterable of ``distributed.Future`` objects,
-        such as come from ``client.submit`` interface. These comprise the individual
+    dfs : Delayed, list[Delayed, Future]
```

Comment:
Maybe this function should also accept a single Future? (In which case, we may need to update the `isinstance(dfs, Delayed)` check on line 686)